stellar-multitenant/pkg/internal/provisioning/manager.go

```diff
@@ -106,9 +128,31 @@ func (m *Manager) ProvisionNewTenant(
 	}
 
 	tenantStatus := tenant.ProvisionedTenantStatus
-	t, err = m.tenantManager.UpdateTenantConfig(ctx, &tenant.TenantUpdate{ID: t.ID, Status: &tenantStatus, SDPUIBaseURL: &uiBaseURL})
+	t, err = m.tenantManager.UpdateTenantConfig(
+		ctx,
+		&tenant.TenantUpdate{
+			ID:                  t.ID,
+			Status:              &tenantStatus,
+			SDPUIBaseURL:        &uiBaseURL,
+			DistributionAccount: &distributionAccPubKey,
+		})
 	if err != nil {
-		return nil, fmt.Errorf("updating tenant %s status to %s: %w", name, tenant.ProvisionedTenantStatus, err)
+		errMsg := fmt.Sprintf("updating tenant %s status to %s: %v", name, tenant.ProvisionedTenantStatus, err)
+		// Rollback distribution account provisioning
+		sigClientDeleteKeyErr := m.distAccSigClient.Delete(ctx, distributionAccPubKey)
+		if sigClientDeleteKeyErr != nil {
+			sigClientDeleteKeyErrMsg := fmt.Sprintf("unable to delete distribution account private key: %v", sigClientDeleteKeyErr)
+			if errors.Is(err, signing.ErrUnsupportedCommand) {
+				log.Ctx(ctx).Warnf(
+					"Private key deletion not needed for distribution account signature client type %s: %v",
+					m.distAccSigClient.Type(), err)
+			} else {
+				log.Ctx(ctx).Error(sigClientDeleteKeyErrMsg)
+				errMsg += fmt.Sprintf(". %s", sigClientDeleteKeyErrMsg)
+			}
+		}
+
+		return nil, errors.New(errMsg)
```

Comment:
👋  To keep the stack trace of the errors, you should use `fmt.Errorf` with the `%w` verb. Feel free to build the custom messages, but keep in mind that any errors would be aggregated using `fmt.Errorf("...: %w")`

More info about the reason for that can be found in https://go.dev/blog/go1.13-errors#errors-in-go-113